async-nats/src/jetstream/message.rs

```diff
@@ -170,52 +166,46 @@ impl Message {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn double_ack(&self) -> Result<(), Error> {
+    pub async fn double_ack(&self) -> Result<(), AckError> {
         if let Some(ref reply) = self.reply {
             let inbox = self.context.client.new_inbox();
-            let mut subscription = self.context.client.subscribe(inbox.clone()).await?;
+            let mut subscription = self
+                .context
+                .client
+                .subscribe(inbox.clone())
+                .map_err(|e| e.with_kind(AckErrorKind::RequestFailed))
+                .await?;
             self.context
                 .client
                 .publish_with_reply(reply.to_string(), inbox, AckKind::Ack.into())
+                .map_err(|e| e.with_kind(AckErrorKind::RequestFailed))
                 .await?;
             match tokio::time::timeout(self.context.timeout, subscription.next())
                 .await
-                .map_err(|_| {
-                    std::io::Error::new(
-                        std::io::ErrorKind::TimedOut,
-                        "double ack response timed out",
-                    )
-                })? {
+                .map_err(|e| e.with_kind(AckErrorKind::TimedOut))?
+            {
                 Some(_) => Ok(()),
-                None => Err(Box::new(std::io::Error::new(
-                    std::io::ErrorKind::Other,
-                    "subscription dropped",
-                ))),
+                None => Err(AckErrorKind::SubscriptionDropped.into()),
```

Comment:
Errors are aggregated into `Other` kind on purpose: to not clutter the possible enum variants with ones that it makes not much senes to match against for the user.